src/DocumentEditor.ts

```diff
@@ -4,154 +4,79 @@
     private ignoreSave: boolean = false;
-    private isLocalDocOpen: boolean = false;
-
-    private recoveredDocsFolder: string;
-    private recoveredFileName: string | undefined;
 
     private readonly dontShowKey: string = 'cosmosDB.dontShow.SaveEqualsUpdateToAzure';
 
-    constructor(context: vscode.ExtensionContext) {
-        // Use a workspace-specific path unique to our extension if possible. Otherwise, just use the main folder for our extension
-        this.localDocPath = path.resolve(path.join(context.storagePath || context.extensionPath, 'cosmos-document.json'));
-        this.recoveredDocsFolder = path.join(context.extensionPath, 'recoveredDocs');
-
-        this.promptForRecoveredDocs();
-    }
-
-    public async showDocument(docNode: IDocumentNode): Promise<void> {
-        // Prompt to update opened doc if it's dirty
-        if (this.lastOpenedDocNode && this.localDoc) {
-            // soft-copy the node and doc to avoid race conditions
-            const doc: vscode.TextDocument = this.localDoc;
-            const node: IDocumentNode = this.lastOpenedDocNode;
-
-            if (doc.isDirty) {
-                const message: string = `Your changes to document '${this.lastOpenedDocNode.label}' will be lost. Update to Azure?`;
-                const result: string | undefined = await vscode.window.showWarningMessage(message, DialogBoxResponses.Yes, DialogBoxResponses.No);
-                if (result === DialogBoxResponses.Yes) {
-                    await this.udpateDocumentToNode(node, doc);
-                } else if (result === undefined) {
-                    throw new UserCancelledError();
-                }
-            }
-        }
+    public async showDocument(docNode: IDocument, fileName: string): Promise<void> {
+        const localDocPath = path.join(os.tmpdir(), randomUtils.getRandomHexString(12), fileName);
+        await fse.ensureFile(localDocPath);
 
-        await fse.ensureFile(this.localDocPath);
-        this.localDoc = await vscode.workspace.openTextDocument(this.localDocPath);
-        await vscode.window.showTextDocument(this.localDoc);
-        this.lastOpenedDocNode = docNode;
-        this.isLocalDocOpen = true;
-        await this.updateEditor(docNode.data);
+        const document = await vscode.workspace.openTextDocument(localDocPath);
+        this.fileMap[localDocPath] = [document, docNode];
+        const textEditor = await vscode.window.showTextDocument(document);
+        await this.updateEditor(docNode.data, textEditor);
     }
 
-    public async updateLastDocument(): Promise<void> {
-        if (this.lastOpenedDocNode && this.localDoc) {
-            await this.udpateDocumentToNode(this.lastOpenedDocNode, this.localDoc);
-        } else {
-            throw new Error('You must select a Document in the CosmosDB explorer before updating to Azure.');
-        }
+    public async updateLastDocument(doc): Promise<void> {
```

Comment:
Rename to `updateMatchingNode` or something. It's not the 'last document' anymore